Malformed packets apply to both TLS and DTLS (#22)
radext-wg/draft-ietf-radext-radiusdtls-bis: draft-ietf-radext-radiusdtls-bis.md

The value of the idle timeout to use depends on the exact deployment and is a trade-of between resource usage on clients/servers and the overhead of opening new connections. Very short timeouts that are at or below the timeouts used for application layer watchdogs, typically in the range of 30-60s can be considered unreasonable. In contrast, the upper limit is much more difficult to define but may be in the range of 10 to 15min, depending on the available resources, or never (disabling idle timeout) in scenarios where a permanently open connection is required.

## Malformed Packets and Unknown clients

The RADIUS specifications say that an implementation should "silently discard" a packet in a number of circumstances.
This action has no further consequences for UDP based transports, as the "next" packet is completely independent of the previous one.

When TLS is used as transport, decoding the "next" packet on a connection depends on the proper decoding of the previous packet.

> As a result the behavior with respect to discarded packets has to change. With DTLS, while the "next" packet does not depend on proper decoding of the previous packet, but since DTLS provides
> integrity protection, any decoding error should be treated as a broken connection.

Reviewer: ```suggestion
As a result the behavior with respect to discarded packets has to change, since a malformed RADIUS packet could impact the decoding of succeeding packets.

With DTLS, the "next" packet does not depend on proper decoding of the previous packet, since the RADIUS packets are sent in independent DTLS records.
However, since both TLS and DTLS provide integrity protection and ensure that the packet was sent by the peer, a protocol violation at this stage implies that the peer is misbehaving.
```

> Implementations of this specification SHOULD tread the "silently discard" texts in the RADIUS specification referenced above as "silently discard and close the connection".

Reviewer: ```suggestion
Implementations of this specification SHOULD read the "silently discard" texts in the RADIUS specification referenced above as "silently discard and close the connection".
```

or

```suggestion
Implementations of this specification SHOULD treat the "silently discard" texts in the RADIUS specification referenced above as "silently discard and close the connection".
```

Reviewer: (Error was the same in the previous version, but now I noticed it.)

> That is, the implementation SHOULD send a TLS close notification and the underlying connection MUST be closed if any of the following circumstances are seen:

Reviewer: ```suggestion
That is, the implementation SHOULD send a TLS close notification and, in the case of RADIUS/TLS, the underlying TCP connection MUST be closed if any of the following circumstances are seen:
```

The original text was specific for RADIUS/TLS, and RADIUS/DTLS does not have a "connection", and closing the UDP socket is an implementation detail.
I don't have a strong opinion about it, but that's what just came to my mind.
